virtdata-userlibs/src/test/java/io/virtdata/IntegratedComposerLogicTest.java

```diff
@@ -136,10 +136,8 @@ public void sanityCheckFunctionCasting() {
         assertThat(ClassUtils.isAssignable(c1, c2)).isTrue();
 
         Class<?> c3 = Identity.class;
-        Class<?> c4 = LongFunction.class;
+        Class<?> c4 = LongUnaryOperator.class;
         assertThat(ClassUtils.isAssignable(c3, c4)).isTrue();
-        LongUnaryOperator f;
-        f = new Identity();
```

Comment:
i could have sworn that i ran the test when i added the missing `isTrue()` but since the tests are failing on the main branch right now, i was probably imagening things.

i went back to a commit before any of my recent changes and this never returned `true` as is... the code below already showed that `Identity` was only ever assignable to `LongUnaryOperator` not `LongFunction`.

Reply:
not sure what is going on but on the main branch sometimes `mvn clean package` passes as well... where as it always fail when run via intellij... at least with this change this code looks more correct in this test? :sweat_smile:

Reply:
I think this is working a little differently depending on the generation of JVM. There have been some subtle changes since the functional types were added, it seems.
I'm just fixing these up to current observed behavior, as it doesn't seem critical path to the other type finesses that are built-in to the current function composer logic.